File: base/src/main/java/com/ssz/baselibrary/app/di/module/AppModule.java
package com.ssz.baselibrary.app.di.module;

import com.ssz.baselibrary.app.AppContext;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author : zsp
 * time : 2019 11 2019/11/19 12:45
 */
@Module
public class AppModule {

    AppContext context;

    public AppModule(AppContext context) {
        this.context = context;
    }

    @Singleton
    @Provides
    public AppContext provideAppContext() {
        return context;
    }
}

File: app/src/main/java/com/ssz/studydemo/base/dagger/di/module/AppModule.kt
package com.ssz.studydemo.base.dagger.di.module

import com.ssz.studydemo.test.app.AppDelegate
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author : zsp
 * time : 2019 12 2019/12/11 14:28
 */
@Module
class AppModule(val appDelegate : AppDelegate) {

    @Singleton
    @Provides
    fun provideAppDelegate(): AppDelegate {
        return appDelegate
    }


}